cmd/keymasterd/config.go

```diff
@@ -571,40 +579,52 @@ func generateCerts(configDir string, config *baseConfig, rsaKeySize int) error {
 	caTemplate.SerialNumber = serialNumber
 	caTemplate.IsCA = true
 	caTemplate.KeyUsage |= x509.KeyUsageCertSign
-	caTemplate.ExtKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth}
+	caTemplate.ExtKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth,
+		x509.ExtKeyUsageServerAuth}
 	caTemplate.Subject = pkix.Name{Organization: []string{"Acme Co CA"}}
-	adminCACertFilename := configDir + "/adminCA.pem"
-	caDer, err := generateCertAndWriteToFile(adminCACertFilename, &caTemplate, &caTemplate, &adminCAKey.PublicKey, adminCAKey)
-	if err != nil {
-		logger.Printf("Failed to create certificate: %s", err)
-		return err
-	}
-	// Now the admin client
-	caCert, err := x509.ParseCertificate(caDer)
-	if err != nil {
-		logger.Printf("Failed to parse certificate: %s", err)
-		return err
-	}
-	clientKeyFilename := configDir + "/adminClient.key"
-	clientKey, err := generateRSAKeyAndSaveInFile(clientKeyFilename, rsaKeySize)
-	if err != nil {
-		logger.Printf("Failed to generate file for key: %s", err)
-		return err
-	}
-	//Fix template!
-	clientTemplate := template
-	//client.KeyUsage |= ExtKeyUsageClientAuth
-	clientTemplate.ExtKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth}
-	clientCertFilename := configDir + "/adminClient.pem"
-	_, err = generateCertAndWriteToFile(clientCertFilename, &clientTemplate, caCert, &clientKey.PublicKey, adminCAKey)
-	if err != nil {
-		logger.Printf("Failed to create certificate: %s", err)
-		return err
+	if needAdminCA {
```

Comment:
Since you are here, why not make all of this a function to simplify this function? (and thanks for the cleanup)

Reply:
Done.